drivers/hmis/app/graphql/mutations/ac_hmis/update_referral_posting.rb

```diff
@@ -0,0 +1,94 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module Mutations
+  class AcHmis::UpdateReferralPosting < CleanBaseMutation
+    description 'Update a referral posting'
+
+    argument :id, ID, required: true
+    argument :input, Types::HmisSchema::ReferralPostingInput, required: false
+
+    field :record, Types::HmisSchema::ReferralPosting, null: true
+    field :errors, [Types::HmisSchema::ValidationError], null: false, resolver: Resolvers::ValidationErrors
+
+    def resolve(id:, input:)
+      handle_error('connection not configured') unless HmisExternalApis::AcHmis::LinkApi.enabled?
+
+      posting = HmisExternalApis::AcHmis::ReferralPosting.active.viewable_by(current_user).find(id)
+      handle_error('referral not found') unless posting
+
+      errors = HmisErrors::Errors.new
+      project = posting.project
+      allowed = current_user.can_manage_incoming_referrals_for?(posting.project)
+      handle_error('access denied') unless allowed
+
+      posting.current_user = current_user
+      posting.attributes = input.to_params
+
```

Comment:
I think we'll want some more validations here:
* if status is Denied Pending, Denial Reason should be set
* if status is Denied, Referral Result and Denial Note should be set (thinking [ahead](https://www.pivotaltracker.com/story/show/185282623)..)


Those can be added like: `errors.add :denial_reason, :required if condition`

The requiredness indicator would need to match, hence https://github.com/greenriver/hmis-frontend/pull/302/files#r1218573843. (When using SubmitForm we only specify the validation on the form definition, since SubmitForm will validate based on that as well. Here we would need to specify it in two places.).

Reply:
> if status is Denied Pending, Denial Reason should be set

This validation is on the model (as you saw). Those validations get added to the errors object, so I think this is already done

> if status is Denied, Referral Result and Denial Note should be set

Thanks! I'd probably make this change as part of 185282623 if it's all the same

Reply:
Oh right, yeah that's fine in this case. Those should get correctly linked to the right input field when displayed to the user.